scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/Evaluation.java

```diff
@@ -156,22 +156,25 @@ public int hashCode() {
     return Objects.hash(goalEvals);
   }
 
-  boolean canAssociateMoreToCreatorOf(ActivityInstance instance){
-    Goal creator = getGoalCreator(instance);
-    if(creator instanceof ActivityExistentialGoal activityExistentialCreator) {
-      //we can piggyback
-      return activityExistentialCreator.childCustody == ChildCustody.Jointly;
-    }
-    return true;
+  boolean canAssociateMoreToCreatorOf(final ActivityInstance instance){
+    final var creator$ = getGoalCreator(instance);
+    if (creator$.isEmpty()) return false;
+    final var creator = creator$.get();
+
+    if (!(creator instanceof ActivityExistentialGoal activityExistentialCreator)) return true;
+    return activityExistentialCreator.childCustody == ChildCustody.Jointly; // we can piggyback
   }
 
-  Goal getGoalCreator(ActivityInstance instance){
-    for(var goalEval : goalEvals.entrySet()){
+  /**
+   * If an activity instance was already in the plan prior to this run of the scheduler, this method will return Optional.empty()
+   */
+  Optional<Goal> getGoalCreator(final ActivityInstance instance){
```

Comment:
Good catch on the missing use case. What you implemented here is a default behavior saying we can piggyback any user-inserted activity. Which seems fine for now but I think this illustrates a missing piece in the server. 
 
When the scheduler gets the plan, it should fetch who created the activity : a goal or a user. And it should get the custody property associated to the activity. If created with a goal, we get the custody from the goal definition. If created by the user, we should get it from the user somehow. Then, all of this is passed to a construction of the Evaluation in the solver.

Reply:
That's a good point... I think I need to better understand this "Custody" concept and some of the motivations behind it. In my mind, the user should be able to manually create activities to satisfy any goal they wish... but perhaps I'm under-thinking things here.

Reply:
Seems like it is outside the scope of this PR.